### PR TITLE
Use default when no request.accept absent

### LINKED ISSUE #24: Doesn't work if 'Accept' header isn't specified
This occurs in controller specs that don't specify an accept header. Maybe we should default to application/json instead of bailing out?

### PR DESCRIPTION
**Pull request by ynnadkrap**
This fixes #24, and also did some refactoring for fun. Let me know what you guys think.

### CODE REVIEW

File: lib/shapeable/shape.rb [hunk 1]
```diff
@@ -4,46 +4,74 @@ module Shape
...
+        raise Shapeable::Errors::UnresolvedVersionError
+      elsif opts[:enforce_shape] && shape.blank?
+        raise Shapeable::Errors::UnresolvedShapeError
+      end
+
+      if opts[:enforce_versioning]
+        construct_constant(opts[:path], shape: shape, version: version)
+      else
+        construct_constant(opts[:path], shape: shape)
       end
```
Comment (ynnadkrap) [inline lib/shapeable/shape.rb]: I think we need a 3rd case here that accounts for `shape` and `version` not being enforced.

File: lib/shapeable/shape.rb [hunk 2]
```diff
@@ -4,46 +4,74 @@ module Shape
...
+          :version,
+          :enforce_versioning,
+          :enforce_shape
+        ].include?(k)
+      end
+    end
+
+    def validate_and_resolve_shape(opts)
+      version = opts[:default_version] || opts[:version]
+      shape = opts[:default_shape] || opts[:shape]
```
Comment (ynnadkrap) [inline lib/shapeable/shape.rb]: Swap `default_x` and `x` to `x` || `default_x`

**Comment by dnasseri** [comment]
@ynnadkrap Looks good to me the only thing I would say is if you could fix the build it would be awesome :). Also, can you bump the version? and we can push a new one to ruby gems and maybe upgrade inside our codebase as well.

**Comment by ynnadkrap** [comment]
On a side note, I think the difference in behavior for `enforce_versioning` and `enforce_shape` is a little strange. 

`When enforce_versioning is set to false, version will be ignored`
`When enforce_shape is set to false, shape will be optional`

**Pull request merged**